src/Track.js

```diff
@@ -5,37 +5,37 @@ import note from './note.jpg'
 
 
 
-const Track = (props) =>{
+const Track = ({song, artist, album, albumCover, isAdd, accessKey, addSong, removeSong}) =>{
     
-    const [trackName, setTrackName] = useState(props.song)
-    const [artist, setArtist] = useState(props.artist)
-    const [album, setAlbum] = useState(props.album)
-    const [albumPic, setAlbumPic] = useState(props.albumCover)
-    const [isAdd, setAdd] = useState(true);
-    const [songId, setSongId] = useState(props.accessKey);
+    // const [trackName, setTrackName] = useState(props.song)
+    // const [artist, setArtist] = useState(props.artist)
+    // const [album, setAlbum] = useState(props.album)
+    // const [albumPic, setAlbumPic] = useState(props.albumCover)
+    // const [isAdd, setAdd] = useState(true);
+    // const [songId, setSongId] = useState(props.accessKey);
    
-useEffect(()=>{
-    if(!props.isAdd){
+// useEffect(()=>{
+//     if(!props.isAdd){
         
-        setAdd(false);
-    }
-},[])
+//         setAdd(false);
+//     }
+// },[props.isAdd])
     
```

Comment:
I removed all of this state logic. If you're not planning on changing any of this information in the Track component, you can just read directly from the props. When you save them again to state, you have to manage all the updating logic yourself (like you figured out with the useEffect).